cmd/eno-controller/main.go

```diff
@@ -65,6 +65,7 @@ func runController() error {
 	flag.StringVar(&synconf.PodNamespace, "synthesizer-pod-namespace", os.Getenv("POD_NAMESPACE"), "Namespace to create synthesizer pods in. Defaults to POD_NAMESPACE.")
 	flag.StringVar(&synconf.ExecutorImage, "executor-image", os.Getenv("EXECUTOR_IMAGE"), "Reference to the image that will be used to execute synthesizers. Defaults to EXECUTOR_IMAGE.")
 	flag.StringVar(&synconf.PodServiceAccount, "synthesizer-pod-service-account", "", "Service account name to be assigned to synthesizer Pods.")
+	flag.DurationVar(&synconf.ContainerCreationTimeout, "container-creation-ttl", time.Second*3, "Timeout when waiting for kubelet to ack scheduled pods. Protects tail latency from kubelet network partitions")
```

Comment:
will 3 seconds too short for kubelet to schedule pod? curious how to define this value used here

Reply:
The timeout only covers the period between when the pod is scheduled and the container statuses are written by kubelet, which normally takes <10ms.

But setting it to something reasonable like 100ms isn't safe since eno-controller and kube-scheduler probably aren't running on the same node so clock skew might be enough to make every pod timeout. So 3s seems like a reasonable tradeoff.

If nodes have clock skew of >3s, Eno will delete the first pod but won't delete the second until the first pod has had time to terminate, which takes a while. So even extreme clock skew is relatively safe.

One other safeguard we could add: disable this timeout after a few retries. That would handle very high clock skew _and_ fast pod termination (unlikely, but theoretically possible). Thoughts?

Reply:
Now that I think about it, that other safeguard sounds like a good idea - just pushed a commit :)